backend-api/src/main/java/pms/pets/api/PetsController.java

```diff
@@ -0,0 +1,38 @@
+package pms.pets.api;
+
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pms.pets.services.PetsServices;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/v1/pets", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+public class PetsController {
+
+    private PetsServices petsServices;
+
+    public PetsController(PetsServices petsServices) {
+        this.petsServices = petsServices;
+    }
+
+    @GetMapping()
+    public ResponseEntity<List<Pet>> get(){
+        var pets = new ArrayList<Pet>();
+        this.petsServices.getAll()
+                        .thenAccept(
+                                lp -> lp.forEach(pet ->
+                                pets.add(new Pet(pet.getName(),
+                                    pet.getId(),
+                                    pet.getRace(),
+                                    pet.getGender().getGenderCode(),
+                                    pet.getAge(),
+                                    pet.getBirthDate()))));
+
+        return ResponseEntity.ok().body(pets);
```

Comment:
We could use the Streams API here (similar to `LINQ` in .NET) and also we can return a future (similar to `Task` in .NET):

```
    public CompletableFuture<ResponseEntity<List<Pet>>> get(){
        return this.petsServices.getAll()
                .thenApplyAsync(
                        lp -> ResponseEntity.ok().body(lp
                                .stream()
                                .map(pet -> new Pet(
                                        pet.getName(),
                                        pet.getId(),
                                        pet.getRace(),
                                        pet.getGender().getGenderCode(),
                                        pet.getAge(),
                                        pet.getBirthDate()))
                                .collect(Collectors.toList())));
```

Reply:
Initially, I used the streams API, but IntelliJ recommended to remove it since it was redundant.
From what I remember I didn't use your approach, that is why it was showing redundancy.

With this approach, we need to be careful with the swagger documentation, since it will show a 200 OK with the CompletableFuture structure. But when testing the Response is in the correct format.

Saying this I will apply this.

Reply:
Hmm then the issue was most likely related to something in your code besides the usage of streams, I tested that snippet in intellij and I didn't see any warning.

Reply:
I saw this which might resolve our issue with `CompletableFuture` showing up in the swagger docs: https://github.com/swagger-api/swagger-codegen/blob/master/samples/server/petstore/spring-mvc-j8-async/src/main/java/io/swagger/api/PetApi.java#L99-L101

Reply:
Yep, I said it was the code 😄 I was using something like `streams().foreach()` which is stupid...

That link is what I am looking for. Saved me a bit of time.

Reply:
:+1: